hack/make/validate-vet

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+source "$(dirname "$BASH_SOURCE")/.validate"
+
+IFS=$'\n'
+files=( $(validate_diff --diff-filter=ACMR --name-only -- '*.go' | grep -v '^vendor/' || true) )
+unset IFS
+
+badFiles=()
+for f in "${files[@]}"; do
+	# we use "git show" here to validate that what's committed passes go vet
+	if [ "$(git show "$VALIDATE_HEAD:$f" | go vet)" ]; then
+		badFiles+=( "$f" )
```

Comment:
Yeah, I'm with @LK4D4; I didn't mean that we should just list the files with problems -- I agree 1000% that we should list the actual problems.  I just want consistency in the script and the output (ie, instead of printing out the list of `badFiles` at the end, print out the actual errors from `go vet`).